Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -616,40 +648,45 @@ namespace Terrain
                     lodIndex2 += LodGridVerts1D;
                 }
 
-                if (lodHeights[lodIndex1] == NoTerrainVertexHeight || lodHeights[lodIndex2] == NoTerrainVertexHeight)
+                uint16_t zIndex = m_vertexOrder.at(index);
+
+                if (lodHeightsNormals[lodIndex1].m_height == NoTerrainVertexHeight || lodHeightsNormals[lodIndex2].m_height == NoTerrainVertexHeight)
                 {
                     // One of the neighboring vertices has no data, so use the original height and normal
-                    clodHeights[index] = originalHeights[index];
-                    clodNormals[index] = originalNormals[index];
+                    clodHeightNormals[zIndex] = originalHeightsNormals[zIndex];
                 }
                 else
                 {
-                    clodHeights[index] = (lodHeights[lodIndex1] + lodHeights[lodIndex2]) / 2;
-                    clodNormals[index].first = (lodNormals[lodIndex1].first + lodNormals[lodIndex2].first) / 2;
-                    clodNormals[index].second = (lodNormals[lodIndex1].second + lodNormals[lodIndex2].second) / 2;
+                    clodHeightNormals[zIndex] =
+                    {
+                        HeightDataType((lodHeightsNormals[lodIndex1].m_height + lodHeightsNormals[lodIndex2].m_height) / 2),
+                        NormalDataType(
+                        {
+                            int16_t((lodHeightsNormals[lodIndex1].m_normal.first + lodHeightsNormals[lodIndex2].m_normal.first) / 2),
+                            int16_t((lodHeightsNormals[lodIndex1].m_normal.second + lodHeightsNormals[lodIndex2].m_normal.second) / 2)
+                        })
+                    };
                 }
             }
         }
 
-        sector.m_lodHeightsBuffer->UpdateData(clodHeights.data(), clodHeights.size() * sizeof(HeightDataType), 0);
-        sector.m_lodNormalsBuffer->UpdateData(clodNormals.data(), clodNormals.size() * sizeof(NormalDataType), 0);
+        sector.m_lodHeightsNormalsBuffer->UpdateData(clodHeightNormals.data(), clodHeightNormals.size() * sizeof(HeightNormalVertex), 0);
     }
 
     void TerrainMeshManager::InitializeCommonSectorData()
     {
         PatchData patchData;
-        InitializeTerrainPatch(GridSize, patchData);
+        InitializeTerrainPatch(patchData);
 
-        m_xyPositionsBuffer = CreateMeshBufferInstance(XYPositionFormat, patchData.m_xyPositions.size(), patchData.m_xyPositions.data());
-        m_indexBuffer = CreateMeshBufferInstance(AZ::RHI::Format::R16_UINT, patchData.m_indices.size(), patchData.m_indices.data());
+        m_xyPositionsBuffer = CreateMeshBufferInstance(AZ::RHI::GetFormatSize(XYPositionFormat), uint32_t(patchData.m_xyPositions.size()), patchData.m_xyPositions.data());
+        m_indexBuffer = CreateMeshBufferInstance(AZ::RHI::GetFormatSize(AZ::RHI::Format::R16_UINT), uint32_t(patchData.m_indices.size()), patchData.m_indices.data());
```

Comment:
Nit: this should use either static_cast<uint32_t> or aznumeric_cast<uint32_t> instead of a c-style cast.